lms/static/js/bookmarks/views/bookmarks_list.js

```diff
@@ -0,0 +1,133 @@
+                this.loadingMessageView = options.loadingMessageView;
+                this.errorMessageView = options.errorMessageView;
+                this.courseId = $(this.el).data('courseId');
+                this.langCode = $(this.el).data('langCode');
+                _.bindAll(this, 'render', 'breadcrumbTrail', 'humanFriendlyDate', 'createBookmarkUrl');
+            },
+
+            render: function () {
+                var data = {
+                    bookmarks: this.collection.models,
+                    breadcrumbTrail: this.breadcrumbTrail,
+                    humanFriendlyDate: this.humanFriendlyDate,
+                    createBookmarkUrl: this.createBookmarkUrl
+                };
+                this.$el.html(this.template(data));
+                this.delegateEvents();
+                return this;
+            },
+
+            showBookmarksList: function () {
+                var view = this;
+
+                this.hideErrorMessage();
+                this.showBookmarksContainer();
+                this.showLoadingMessage();
+
+                this.collection.url = this.url;
+                this.collection.fetch({
+                    reset: true,
+                    data: {course_id: this.courseId, fields: 'display_name,path'}
+                }).done(function () {
+                    view.hideLoadingMessage();
+                    view.render();
+                    view.focusBookmarksElement();
+                }).fail(function () {
+                    view.hideLoadingMessage();
+                    view.showErrorMessage();
+                });
+            },
+
+            visitBookmark: function (event) {
+                window.location = event.target.pathname;
+            },
+
+            /**
+             * Create a breadcrumb trail from section, subsection and unit display names.
+             * @param {Array} bookmarkPath - Contains 2 objects of info related to section and subsection.
+             * @param {String} unitDisplayName - Unit display name.
+             */
+            breadcrumbTrail: function (bookmarkPath, unitDisplayName) {
+                var separator = ' <i class="icon fa fa-caret-right" aria-hidden="true"></i><span class="sr">-</span> ';
+                var names = _.pluck(bookmarkPath, 'display_name');
+                names.push(unitDisplayName);
+                return names.join(separator);
+            },
+
+            /**
+             * Convert ISO 8601 formatted date into human friendly format. e.g, `2014-05-23T14:00:00Z` to `May 23, 2014`
+             * @param {String} isoDate - ISO 8601 formatted date string.
+             */
+            humanFriendlyDate: function (isoDate) {
+                moment.locale(this.langCode);
+                return moment(isoDate).format('LL');
+            },
+
+            createBookmarkUrl: function (courseId, usageId) {
+                return '/courses/' + courseId + '/jump_to/' + usageId
+            },
```

Comment:
This is information about the bookmark object so it would make more sense to make it a property of the model. `blockUrl` would be a nice name.